index.js

```diff
@@ -213,16 +213,16 @@ class Analytics {
 
     if (!this.flushed) {
       this.flushed = true
-      this.flush()
+      this.flush(callback)
       return
     }
 
     if (this.queue.length >= this.flushAt) {
-      this.flush()
+      this.flush(callback)
```

Comment:
Weird bug, may not cause issues, but this actually causes the callback to be called twice now as its the exact same callback pulled from the item when flushed.
https://github.com/segmentio/analytics-node/blob/master/index.js#L247-L259

Reply:
cc - @pbassut

Reply:
Yes, the callback is running twice for me as well. Is there a plan to fix that? Or should I log a new issue?

Reply:
We should have a fix ASAP

Reply:
Hi. Was the fix for this released with v5.1.0?

Reply:
@pbassut could you please check if this is already fixed?

Reply:
@pooyaj It's fixed locally. I'm writing some tests and I'll create a PR for it.

Reply:
@BrandonNoad Can you give me a minimal working example that illustrates the issue? 

For issue `#1`: seems unlikely given they're only called when the request to segment servers actually finishes(successfully or not).

As for issue `#2`: I'll have to see some code.

Reply:
@BrandonNoad Please open a new issue for it though.

Reply:
@pbassut Yes, I can open a new issue with some code to reproduce the behaviour.

> For issue #1: seems unlikely given they're only called when the request to segment servers actually finishes(successfully or not).

I may have been incorrect in saying "too early". That was just an assumption based on the fact that the callback is being called twice and in the first call, `batch` is `undefined`.

Reply:
New issue is here: https://github.com/segmentio/analytics-node/issues/300